internal/provisioner/rtcd.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package provisioner
+
+import (
+	"strings"
+
+	"github.com/mattermost/mattermost-cloud/internal/tools/aws"
+	"github.com/mattermost/mattermost-cloud/internal/tools/kops"
+	"github.com/mattermost/mattermost-cloud/model"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type rtcd struct {
+	awsClient      aws.AWS
+	environment    string
+	provisioner    *KopsProvisioner
+	kops           *kops.Cmd
+	cluster        *model.Cluster
+	logger         log.FieldLogger
+	desiredVersion *model.HelmUtilityVersion
+	actualVersion  *model.HelmUtilityVersion
+}
+
+func newRtcdHandle(cluster *model.Cluster, desiredVersion *model.HelmUtilityVersion, provisioner *KopsProvisioner, awsClient aws.AWS, kops *kops.Cmd, logger log.FieldLogger) (*rtcd, error) {
+	if logger == nil {
+		return nil, errors.New("cannot instantiate RTCD handle with nil logger")
+	}
+
+	if provisioner == nil {
+		return nil, errors.New("cannot create a connection to RTCD if the provisioner provided is nil")
+	}
+
+	if kops == nil {
+		return nil, errors.New("cannot create a connection to PgbRTCDouncer if the Kops command provided is nil")
+	}
+
```

Comment:
I think we should do more validation here, specifically on `awsClient` and `kops`, see kubecost for example: https://github.com/mattermost/mattermost-cloud/blob/rtcd-changes-angelos/internal/provisioner/kubecost.go#L47